cognite/neat/workflows/steps/lib/rules_exporter.py

```diff
@@ -49,9 +77,71 @@ def run(self, transformation_rules: RulesData) -> (FlowMessage, DMSDataModel):
             f"</ul> Data model is meant to be uploaded to <b><code>{data_model.space}</code></b> space"
         )
 
+        # need to store the data model in the step so that it can be used by the next step
+        # see GraphQL step
+
         return FlowMessage(output_text=output_text), DMSDataModel(data_model=data_model)
 
 
+class ExportDMSDataModel(Step):
+    """
+    This step exports DMS data model and its building components to YAML files
+    """
+
+    description = "This step generates GraphQL schema from data model defined in transformation rules."
+    category = CATEGORY
+    configurables: ClassVar[list[Configurable]] = [
+        Configurable(name="storage_dir", value="staging", label="Directory to store DMS schema files"),
+    ]
+
+    def run(self, data_model_contract: DMSDataModel) -> FlowMessage:  # type: ignore[override, syntax]
+        if self.configs is None or self.data_store_path is None:
+            raise StepNotInitialized(type(self).__name__)
+
+        staging_dir_str = self.configs["storage_dir"]
+        staging_dir = self.data_store_path / Path(staging_dir_str)
+        staging_dir.mkdir(parents=True, exist_ok=True)
+
+        base_file_name = (
+            f"{data_model_contract.data_model.space}-"
+            f"{data_model_contract.data_model.external_id}-"
+            f"v{data_model_contract.data_model.version.strip().replace('.', '_')}"
+        )
+
+        _container_file_name = f"{base_file_name}-containers.yaml"
+        _data_model_file_name = f"{base_file_name}-data-model.yaml"
+
+        container_full_path = staging_dir / _container_file_name
+        data_model_full_path = staging_dir / _data_model_file_name
+
+        data_model = dm.DataModelApply(
+            space=data_model_contract.data_model.space,
+            external_id=data_model_contract.data_model.external_id,
+            version=data_model_contract.data_model.version,
+            description=data_model_contract.data_model.description,
+            name=data_model_contract.data_model.name,
+            views=list(data_model_contract.data_model.views.values()),
+        )
+
+        containers = dm.ContainerApplyList(data_model_contract.data_model.containers.values())
+
+        container_full_path.write_text(containers.dump_yaml())
+        data_model_full_path.write_text(data_model.dump_yaml())
```

Comment:
Is it a requirement that we use as few files as possible when it is dumped? 

When working with data models as YAML files my experience is that they quickly become very large and hard to maintain. So suggest we follow the standard from `cognite-toolkit` modules of one file per container, per view, and the data model is only specifying which view ids it expects. In addition, we use the prefix, `.container.yaml`, `.view.yaml`, and `data_model.yaml` to indicate what type of resource this is.

See for an example of what I suggest for output. https://github.com/cognitedata/cdf-project-templates/tree/main/cognite_toolkit/cognite_modules/examples/cdf_apm_simple_data_model/data_models

Reply:
Great! I will update it accordingly !